File: Examples/UI/WinForm/CSharp/ListBoxes/NFontListBoxUC.cs
using System;
using System.Collections;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Windows.Forms;

using Nevron.UI.WinForm.Controls;
using Nevron.Examples.Framework.WinForm;

namespace Nevron.Examples.UI.WinForm
{
	/// <summary>
	/// Summary description for NFontListBoxUC.
	/// </summary>
	public class NFontListBoxUC : NExampleUserControl
	{
		#region Constructor

		public NFontListBoxUC(MainForm f) : base(f)
		{
			InitializeComponent();

			Dock = DockStyle.Fill;
		}


		#endregion

		#region Overrides

		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		public override void Initialize()
		{
			base.Initialize ();

			nListBox1.DisplayStyle = FontDisplayStyle.Name;
			nListBox2.DisplayStyle = FontDisplayStyle.NameInFont;
			nListBox3.DisplayStyle = FontDisplayStyle.NameAndSample;
		}


		#endregion

		#region Component Designer generated code
		/// <summary> 
		/// Required method for Designer support - do not modify 
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.nListBox1 = new NFontListBox();
			this.label1 = new System.Windows.Forms.Label();
			this.label2 = new System.Windows.Forms.Label();
			this.nListBox2 = new NFontListBox();
			this.label3 = new System.Windows.Forms.Label();
			this.nListBox3 = new NFontListBox();
			this.SuspendLayout();
			// 
			// nListBox1
			// 
			this.nListBox1.Location = new System.Drawing.Point(16, 32);
			this.nListBox1.Name = "nListBox1";
			this.nListBox1.Size = new System.Drawing.Size(224, 184);
			this.nListBox1.TabIndex = 0;
			// 
			// label1
			// 
			this.label1.Location = new System.Drawing.Point(16, 8);
			this.label1.Name = "label1";
			this.label1.Size = new System.Drawing.Size(224, 23);
			this.label1.TabIndex = 1;
			this.label1.Text = "Name only:";
			this.label1.TextAlign = System.Drawing.ContentAlignment.MiddleLeft;
			// 
			// label2
			// 
			this.label2.Location = new System.Drawing.Point(256, 8);
			this.label2.Name = "label2";
			this.label2.Size = new System.Drawing.Size(256, 23);
			this.label2.TabIndex = 3;
			this.label2.Text = "Name in font:";
			this.label2.TextAlign = System.Drawing.ContentAlignment.MiddleLeft;
			// 
			// nListBox2
			// 
			this.nListBox2.Location = new System.Drawing.Point(256, 32);
			this.nListBox2.Name = "nListBox2";
			this.nListBox2.Size = new System.Drawing.Size(256, 184);
			this.nListBox2.TabIndex = 2;
			// 
			// label3
			// 
			this.label3.Location = new System.Drawing.Point(16, 224);
			this.label3.Name = "label3";
			this.label3.Size = new System.Drawing.Size(264, 23);
			this.label3.TabIndex = 5;
			this.label3.Text = "Name and sample:";
			this.label3.TextAlign = System.Drawing.ContentAlignment.MiddleLeft;
			// 
			// nListBox3
			// 
			this.nListBox3.Location = new System.Drawing.Point(16, 248);
			this.nListBox3.Name = "nListBox3";
			this.nListBox3.Size = new System.Drawing.Size(496, 184);
			this.nListBox3.TabIndex = 4;
			// 
			// NFontListBoxUC
			// 
			this.AutoScroll = true;
			this.Controls.Add(this.label3);
			this.Controls.Add(this.nListBox3);
			this.Controls.Add(this.label2);
			this.Controls.Add(this.nListBox2);
			this.Controls.Add(this.label1);
			this.Controls.Add(this.nListBox1);
			this.Name = "NFontListBoxUC";
			this.Size = new System.Drawing.Size(528, 456);
			this.ResumeLayout(false);

		}
		#endregion

		#region Fields

		private System.ComponentModel.Container components = null;
		private Nevron.UI.WinForm.Controls.NFontListBox nListBox1;
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.Label label2;
		private Nevron.UI.WinForm.Controls.NFontListBox nListBox2;
		private System.Windows.Forms.Label label3;
		private Nevron.UI.WinForm.Controls.NFontListBox nListBox3;

		#endregion
	}
}

File: Examples/UI/WinForm/VB.NET/ListBoxes/NFontListBoxUC.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Windows.Forms

Imports Nevron.UI.WinForm.Controls
Imports Nevron.Examples.Framework.WinForm

Namespace Nevron.Examples.UI.WinForm
	''' <summary>
	''' Summary description for NFontListBoxUC.
	''' </summary>
	Public Class NFontListBoxUC
		Inherits NExampleUserControl
		#Region "Constructor"

		Public Sub New(ByVal f As MainForm)
			MyBase.New(f)
			InitializeComponent()

			Dock = DockStyle.Fill
		End Sub


		#End Region

		#Region "Overrides"

		Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If Not components Is Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub

		Public Overrides Sub Initialize()
			MyBase.Initialize ()

			nListBox1.DisplayStyle = FontDisplayStyle.Name
			nListBox2.DisplayStyle = FontDisplayStyle.NameInFont
			nListBox3.DisplayStyle = FontDisplayStyle.NameAndSample
		End Sub


		#End Region

		#Region "Component Designer generated code"
		''' <summary> 
		''' Required method for Designer support - do not modify 
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			Me.nListBox1 = New NFontListBox()
			Me.label1 = New System.Windows.Forms.Label()
			Me.label2 = New System.Windows.Forms.Label()
			Me.nListBox2 = New NFontListBox()
			Me.label3 = New System.Windows.Forms.Label()
			Me.nListBox3 = New NFontListBox()
			Me.SuspendLayout()
			' 
			' nListBox1
			' 
			Me.nListBox1.Location = New System.Drawing.Point(16, 32)
			Me.nListBox1.Name = "nListBox1"
			Me.nListBox1.Size = New System.Drawing.Size(224, 184)
			Me.nListBox1.TabIndex = 0
			' 
			' label1
			' 
			Me.label1.Location = New System.Drawing.Point(16, 8)
			Me.label1.Name = "label1"
			Me.label1.Size = New System.Drawing.Size(224, 23)
			Me.label1.TabIndex = 1
			Me.label1.Text = "Name only:"
			Me.label1.TextAlign = System.Drawing.ContentAlignment.MiddleLeft
			' 
			' label2
			' 
			Me.label2.Location = New System.Drawing.Point(256, 8)
			Me.label2.Name = "label2"
			Me.label2.Size = New System.Drawing.Size(256, 23)
			Me.label2.TabIndex = 3
			Me.label2.Text = "Name in font:"
			Me.label2.TextAlign = System.Drawing.ContentAlignment.MiddleLeft
			' 
			' nListBox2
			' 
			Me.nListBox2.Location = New System.Drawing.Point(256, 32)
			Me.nListBox2.Name = "nListBox2"
			Me.nListBox2.Size = New System.Drawing.Size(256, 184)
			Me.nListBox2.TabIndex = 2
			' 
			' label3
			' 
			Me.label3.Location = New System.Drawing.Point(16, 224)
			Me.label3.Name = "label3"
			Me.label3.Size = New System.Drawing.Size(264, 23)
			Me.label3.TabIndex = 5
			Me.label3.Text = "Name and sample:"
			Me.label3.TextAlign = System.Drawing.ContentAlignment.MiddleLeft
			' 
			' nListBox3
			' 
			Me.nListBox3.Location = New System.Drawing.Point(16, 248)
			Me.nListBox3.Name = "nListBox3"
			Me.nListBox3.Size = New System.Drawing.Size(496, 184)
			Me.nListBox3.TabIndex = 4
			' 
			' NFontListBoxUC
			' 
			Me.AutoScroll = True
			Me.Controls.Add(Me.label3)
			Me.Controls.Add(Me.nListBox3)
			Me.Controls.Add(Me.label2)
			Me.Controls.Add(Me.nListBox2)
			Me.Controls.Add(Me.label1)
			Me.Controls.Add(Me.nListBox1)
			Me.Name = "NFontListBoxUC"
			Me.Size = New System.Drawing.Size(528, 456)
			Me.ResumeLayout(False)

		End Sub
		#End Region

		#Region "Fields"

		Private components As System.ComponentModel.Container = Nothing
		Private nListBox1 As Nevron.UI.WinForm.Controls.NFontListBox
		Private label1 As System.Windows.Forms.Label
		Private label2 As System.Windows.Forms.Label
		Private nListBox2 As Nevron.UI.WinForm.Controls.NFontListBox
		Private label3 As System.Windows.Forms.Label
		Private nListBox3 As Nevron.UI.WinForm.Controls.NFontListBox

		#End Region
	End Class
End Namespace
